3rdparty/SConscript

```diff
@@ -306,6 +307,28 @@ elif 'sox' in system_dependencies:
 
     env = conf.Finish()
 
+# dep: openssl
+if 'openssl' in autobuild_dependencies:
+    env.BuildThirdParty(thirdparty_build_dir, meta.toolchain,
+        meta.thirdparty_variant, thirdparty_versions, 'openssl')
+elif 'openssl' in system_dependencies:
+    pkg_config_path = None
+    # set custom PKG_CONFIG_PATH in MacOS
+    if meta.platform == 'darwin' and not is_crosscompiling:
+        pkg_config_path = env.FindBrewLibrary('openssl')
```

Comment:
A few minor suggestions:

1. Let's make FindBrewLibrary to be method of config, and invoke conf.FindBrewLibrary instead of env.FindBrewLibrary. The implementation will be moved to config.py in scons plugin. It collects methods for finding system libs, dirs, etc.

2. Let's rename it to FindBrewPackage.

3. Let's make FindBrewPackage a bit more generic: it will return package path, instead of pkgconfig path instead of the package, i.e. it will exclude `'/lib/pkgconfig` from returned path.

4. Accordingly, let's AddPkgConfigDependency handle this more generic version seamlessly. Instead of config_path argument, it will have an argument like prepend_prefix or additional_prefix. AddPkgConfigDependency will treat it as additional installation prefix that should be searched for `/lib/pkgconfig` subdir.

With this changes, FindBrewPackage won't be limited to only libs and to only packages that have pkgconfig files.

Reply:
done